src/NServiceBus.RabbitMQ/Receiving/MessagePump.cs

```diff
@@ -214,6 +214,7 @@ void Connection_ConnectionShutdown(object sender, ShutdownEventArgs e)
                 var errorHandled = false;
                 var numberOfDeliveryAttempts = 0;
 
+                contextBag.Set(message);
```

Comment:
> you'll have to also undo the optimization currently in place around the creation of the ContextBag

how does that work when other things place things in the context? eg https://docs.particular.net/nservicebus/sql-persistence/#sqlstoragesession